### PR TITLE
Fix Random not working for trivial group

### LINKED ISSUE #59: Random doesn't work on trivial PcpGroup
For most of the common types of groups, calling Random on the trivial group works fine. Not so for PcpGroups:
```
gap> Random(TrivialGroup(IsPcGroup));
<identity> of ...
gap> Random(TrivialGroup(IsPermGroup));
()
gap> Random(TrivialGroup(IsFpGroup));
<identity ...>
gap> Random(TrivialGroup(IsPcpGroup));
Error, cannot compute this
 at /usr/lib/gap-4.11.1/pkg/polycyclic/gap/pcpgrp/general.gi:28 called from
MappedVector( g, pcp ) at /usr/lib/gap-4.11.1/pkg/polycyclic/gap/basic/pcpgrps.gi:111 called from
func( GlobalMersenneTwister, x ) at /usr/lib/gap-4.11.1/lib/random.gi:303 called from
<function "Random for a pcp group">( <arguments> )
```
The error happens because `Random` ends up calling `MappedVector( [], Pcp [  ] with orders [  ] )`, and `MappedVector` returns this error when the first argument has zero length. By contrast, the following does work, even though this is again the trivial group:
```
gap> H := AbelianPcpGroup( [ 1 ] );
Pcp-group with orders [ 1 ]
gap> Random( H );
id
```

Would adding the following code to `Random`  in /basic/pcpgrps.gi (line 94) be an acceptable fix?
```
if Length(pcp) = 0 then
  Return One( G );
fi;
```

### PR DESCRIPTION
**Pull request by sTertooy**
Fixes #59.

### CODE REVIEW

File: gap/basic/pcpgrps.gi [hunk 1]
```diff
@@ -99,16 +99,20 @@ InstallMethodWithRandomSource( Random, "for a random source and a pcp group",
...
-        if rel[i] = 0 then
-            g[i] := Random( rs, Integers );
-        else
-            g[i] := Random( rs, 0, rel[i]-1 );
-        fi;
-    od;
-    return MappedVector( g, pcp );
+    if Length( pcp ) = 0 then
+        return One( G );
+    else
```
Comment (fingolfin) [inline gap/basic/pcpgrps.gi:104]:
```suggestion
    if Length( pcp ) = 0 then
        return One( G );
    fi;
```
I would prefer if you just inserted this check, and then left the rest of the function unchanged: Besides reducing the diff (and avoiding issues with incorrect indentation as in the next line), IMHO every level of nesting makes code harder to understand.

**Review by fingolfin — CHANGES_REQUESTED** [review]
Thank you!

I'd prefer if this was done very slightly differently. If you are willing to follow my suggestion, that'd be awesome; otherwise, I'll take care of it later

**Comment by sTertooy** [comment]
Should be fixed now :)

**Review by fingolfin — APPROVED** [review]
Thanks!

**Pull request merged**